src/net/uds/stream.rs

```diff
@@ -0,0 +1,159 @@
+use crate::event::Source;
+use crate::unix::SourceFd;
+use crate::{sys, Interests, Registry, Token};
+
+use std::io::{self, IoSlice, IoSliceMut};
+use std::net::Shutdown;
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+use std::os::unix::net::SocketAddr;
+use std::path::Path;
+
+/// A non-blocking Unix stream socket.
+#[derive(Debug)]
+pub struct UnixStream {
+    std: std::os::unix::net::UnixStream,
+}
+
+impl UnixStream {
+    /// Connects to the socket named by `path`.
+    pub fn connect<P: AsRef<Path>>(p: P) -> io::Result<UnixStream> {
+        let std = sys::uds::connect_stream(p.as_ref())?;
+        Ok(UnixStream { std })
+    }
+
+    /// Converts a `std` `UnixStream` to a `mio` `UnixStream`.
+    ///
+    /// The caller is responsible for ensuring that the socket is in
+    /// non-blocking mode.
+    pub fn from_std(std: std::os::unix::net::UnixStream) -> UnixStream {
```

Comment:
Same as for `UnixDatagram` and `UnixListener` please remove this method and move the docs to `FromRawFd`.